rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -166,11 +166,9 @@ rmw_fastrtps_shared_cpp::create_participant(
 
   // No custom handling of RMW_DEFAULT_DOMAIN_ID. Simply use a reasonable domain id.
 #if FASTRTPS_VERSION_MAJOR < 2
-  participantAttrs.rtps.builtin.domainId =
-    static_cast<uint32_t>(domain_id != RMW_DEFAULT_DOMAIN_ID ? domain_id : 0u);
+  participantAttrs.rtps.builtin.domainId = domain_id;
```

Comment:
@ivanpauno here and below, consider keeping those `static_cast`s.

Reply:
Good point, I didn't see them.